### ISSUE TITLE
Peephole connections (Wci, Wcf, Wco) gradient update

### ISSUE THREAD

**Issue by Pozimek** [automan000/Convolution_LSTM_PyTorch#20]
[The LSTM paper](ftp://ftp.idsia.ch/pub/juergen/TimeCount-IJCNN2000.pdf) defines [a specific rule](https://i.imgur.com/peOKqkL.png) for gradient updates of the 'peephole' connections. Specifically:

>[...] during learning no error signals are propagated back from gates via peephole connections to CEC

Based on my understanding of the code the way these 3 variables are initialized (as asked in [Issue 17](https://github.com/automan000/Convolution_LSTM_PyTorch/issues/17)) is an attempt at implementing this update rule, but I don't see how does initializing them as Variables helps. From my understanding of the quoted part of the LSTM paper, the peephole connections should be updated but the gradient that updates them should stop there and not flow any further. If that is the case then this implementation is incorrect, although it might be that Pytorch does not support such an operation as .detach() is not suitable for the job.

**Comment by Pozimek** [comment]
On a second thought, changing [lines 33](https://github.com/automan000/Convolution_LSTM_PyTorch/blob/master/convolution_lstm.py#L33),34,36 to use detach() on c might work. L33 would look like this:

`        ci = torch.sigmoid(self.Wxi(x) + self.Whi(h) + c.detach() * self.Wci)`

**Comment by Pozimek** [comment]
I've come to think that changing L33, L34 and L36 to use c.detach() should fix this issue, but I'm not very confident about this.

`        ci = torch.sigmoid(self.Wxi(x) + self.Whi(h) + c.detach() * self.Wci)`

IMO gradient should flow through c only via the operations in [L35](https://github.com/automan000/Convolution_LSTM_PyTorch/blob/master/convolution_lstm.py#L35) and L37.
